types/wlr_output_layout.c

```diff
@@ -153,6 +156,8 @@ static struct wlr_output_layout_output *wlr_output_layout_output_create(
 	wl_signal_add(&output->events.destroy, &l_output->state->output_destroy);
 	l_output->state->output_destroy.notify = handle_output_destroy;
 
+	wl_signal_emit(&layout->events.add, l_output);
```

Comment:
If you add auto, the layout output won't be configured here. You have to emit all the events after `wlr_output_layout_reconfigure()` which is guaranteed to be called when the layout changes.

Reply:
In `wlr_output_layout_reconfigure` I don't know which output has been added, which is needed for creating one cursor per output in `wlr_cursor`. I'll instead emit the `add` event in auto.

Reply:
In the end, `wlr_output_layout_add_auto` is also calling `wlr_output_layout_output_create` so it's fine. I also checked that `wlr_output_layout_output` is not created somewhere else too.

Reply:
It's still not right. At this point, the x/y coordinates are always zero. If it gets added auto, those will not be the final coordinates. You should emit the event after the layout is reconfigured.

Reply:
Oh, I misunderstood what you pointed out. It doesn't matter here, but it makes sense to emit the event after the output is fully configured.